syntax = "proto3";

package AccountService_;

// 生成的代码中包含服务
option cc_generic_services = true;

// 有关登录的消息
message LoginRequest
{
    int32 id = 1;
    bytes password = 2;
    bytes msg = 3;
}

message LoginResponse
{
    bool is_success = 1;
    int32 id = 2;
    bytes msg = 3;
    repeated bytes offline_msg = 4;
    bytes name = 5;
}

// 有关注册
message RegisterRequest
{
    bytes name = 1;
    bytes password = 2;
    bytes msg = 3;
}

message RegisterResponse
{
    bool is_success = 1;
    int32 id = 2;
    bytes msg = 3;
}

//有关下线消息
message LogoutRequest
{
    int32 id = 1;
}

message LogoutResponse
{
    bool is_success = 1;
    int32 id = 2;
    bytes msg = 3;
}

message User
{
    int32 id = 1;
    bytes name = 2;
    bytes password = 3;
    repeated int32 friends = 4;
    repeated bytes offline_msg = 5;
}



//服务接口
service AccountServiceRpc
{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Register(RegisterRequest) returns(RegisterResponse);
    rpc Logout (LogoutRequest)returns (LogoutResponse);
}